webcomponents/tool/src/main/frontend/js/grader/sakai-grader.js

```diff
@@ -579,53 +595,134 @@ export class SakaiGrader extends gradableDataMixin(SakaiElement) {
   }
 
   toggleFeedback() {
-
     const feedbackPanel = $("#feedback-panel");
 
     if (!feedbackPanel.dialog("instance")) {
       feedbackPanel.dialog({
         width: "auto",
-        close: () => this.toggleFeedback()
+        beforeClose: () => {
+          const isUnsaved = this.isFeedbackUnsaved();
+          if (isUnsaved) {
+            this.cancelFeedbackToogle();
+          }
+          return !isUnsaved;
+        },
+        close: () => this.cancelFeedbackToogle()
       });
       this.feedbackCommentEditor = this.replaceWithEditor("grader-feedback-comment");
-    } else {
-      this.submission.feedbackComment = this.feedbackCommentEditor.getData();
+      this.previousFeedback = this.submission.feedbackComment;
+    }
+    else if (this.isFeedbackUnsaved()) {
+      this.toggleConfirm("confirm-feedback-panel");
+    }
+    else {
       this.feedbackCommentEditor.destroy();
       feedbackPanel.dialog("destroy");
       this.requestUpdate();
     }
   }
 
   doneWithFeedbackDialog() {
-
+    this.submission.feedbackComment = this.feedbackCommentEditor.getData();
+    this.previousFeedback = this.submission.feedbackComment;
+    this.modified = true;
     this.toggleFeedback();
     document.getElementById("grader-feedback-button").focus();
   }
 
-  togglePrivateNotes() {
+  isFeedbackUnsaved() {
```

Comment:
I don't think you should need this. Look at the way the modified (dirty) flag is throughout the code.

Reply:
Now I change to use dirty! I made some local tests, hope is fine now too.